umi_alignment/workflow.cwl

```diff
@@ -0,0 +1,110 @@
+#!/usr/bin/env cwl-runner
+
+cwlVersion: v1.0
+class: Workflow
+label: "umi alignment workflow"
+requirements:
+    - class: SubworkflowFeatureRequirement
+inputs:
+    read1_fastq:
+        type: File
+    read2_fastq:
+        type: File
+    sample_name:
+        type: string
+    library_name:
+        type: string
+    platform_unit:
+        type: string
+    platform:
+        type: string
+    read_structure:
+        type: string[]
+    reference:
+        type: string
+    target_intervals:
+       type: File?
+outputs:
+    aligned_bam:
+        type: File
+        secondaryFiles: [.bai]
+        outputSource: clip_overlap/clipped_bam
+    adapter_histogram:
+        type: File
+        outputSource: mark_illumina_adapters/metrics
+    duplex_seq_metrics:
+        type: File[]
+        outputSource: collect_duplex_seq_metrics/duplex_seq_metrics
+steps:
+    fastq_to_bam:
+        run: fastq_to_bam.cwl
+        in:
+            read1_fastq: read1_fastq
+            read2_fastq: read2_fastq
+            sample_name: sample_name
+            library_name: library_name
+            platform_unit: platform_unit
+            platform: platform
+        out:
+            [bam]
+    extract_umis:
+        run: extract_umis.cwl
+        in:
+            bam: fastq_to_bam/bam
+            read_structure: read_structure
+        out:
+            [umi_extracted_bam]
+    mark_illumina_adapters:
+        run: mark_illumina_adapters.cwl
+        in:
+            bam: extract_umis/umi_extracted_bam
+        out:
+            [marked_bam, metrics]
+    align:
+        run: align.cwl
+        in:
+            bam: mark_illumina_adapters/marked_bam
+            reference: reference
+        out:
+            [aligned_bam]
+    group_reads_by_umi:
+        run: group_reads.cwl
+        in:
+            bam: align/aligned_bam
+        out:
+            [grouped_bam]
+    call_duplex_consensus:
+        run: call_duplex_consensus.cwl
+        in:
+            bam: group_reads_by_umi/grouped_bam
+        out:
+            [consensus_bam]
+    align_consensus:
+        run: realign.cwl
+        in:
+            bam: call_duplex_consensus/consensus_bam
+            reference: reference
+        out:
+            [consensus_aligned_bam]
+    filter_consensus:
+        run: filter_consensus.cwl
+        in:
+            bam: align_consensus/consensus_aligned_bam
```

Comment:
I would add this as an output file until we have validated this workflow on real data.